### PR TITLE
[Model Monitoring] Delete `ModelMonitoringApplicationBaseV1`

### PR DESCRIPTION
**Pull request by davesh0812**
Delete `ModelMonitoringApplicationBase` and  rename `ModelMonitoringApplicationBaseV2` to `ModelMonitoringApplicationBase`.

related to [ML-7721](https://iguazio.atlassian.net/browse/ML-7721)

[ML-7721]: https://iguazio.atlassian.net/browse/ML-7721?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: tests/system/model_monitoring/test_app.py [hunk 1]
```diff
@@ -40,34 +40,29 @@
...
-    CustomEvidentlyMonitoringAppV2,
 )
 
 
 @dataclass
 class _AppData:
-    class_: type[
-        typing.Union[ModelMonitoringApplicationBase, ModelMonitoringApplicationBaseV2]
-    ]
+    class_: type[typing.Union[ModelMonitoringApplicationBase]]
```
Comment (jond01) [inline tests/system/model_monitoring/test_app.py]:
```suggestion
    class_: type[ModelMonitoringApplicationBase]
```

File: tests/system/model_monitoring/assets/custom_evidently_app.py [hunk 2]
```diff
@@ -130,112 +128,6 @@ def _create_evidently_project(
...
-            tests=[DataDriftTestPreset()],
-            timestamp=schedule_time,
-        )
-
-        data_drift_test_suite.run(reference_data=self.train_set, current_data=sample_df)
-        return data_drift_test_suite
-
-
-class CustomEvidentlyMonitoringAppV2(EvidentlyModelMonitoringApplicationBaseV2):
     NAME = "evidently-app-test-v2"
```
Comment (jond01) [inline tests/system/model_monitoring/assets/custom_evidently_app.py]:
```suggestion
    NAME = "evidently-app-test"
```

File: tests/system/model_monitoring/assets/application.py [hunk 3]
```diff
@@ -33,50 +31,6 @@
...
-                name="model_perf",
-                value=80,
-                kind=ResultKindApp.model_performance,
-                status=ResultStatusApp.no_detection,
-            ),
-        ]
-
-
-class DemoMonitoringAppV2(ModelMonitoringApplicationBaseV2):
     NAME = "monitoring-test-v2"
```
Comment (jond01) [inline tests/system/model_monitoring/assets/application.py]:
```suggestion
    NAME = "monitoring-test"
```

File: mlrun/db/httpdb.py [hunk 4]
```diff
@@ -525,10 +525,6 @@ def connect(self, secrets=None):
                 server_cfg.get("external_platform_tracking")
                 or config.external_platform_tracking
             )
-            config.model_endpoint_monitoring.store_type = (
```
Comment (jond01) [inline mlrun/db/httpdb.py]: Does it have an implication for old clients?

File: docs/tutorials/src/evidently_app.py
Comment (jond01) [inline docs/tutorials/src/evidently_app.py:1]: Just FYI: I have some comments on this, but it's going to be rewritten in [ML-7274](https://iguazio.atlassian.net/browse/ML-7274), so I'll update it there

File: mlrun/model_monitoring/controller.py
Comment (jond01) [inline mlrun/model_monitoring/controller.py:1]: What a lovely rebase for https://github.com/mlrun/mlrun/pull/6191 🙈

File: mlrun/model_monitoring/controller.py [hunk 5]
```diff
@@ -443,154 +418,71 @@ def model_endpoint_process(
...
-                            feature_set=m_fs,
+                    prediction_metric = tsdb_connector.read_predictions(
+                        endpoint_id=endpoint_id,
+                        start=start_infer_time,
+                        end=end_infer_time,
+                    )
+                    if isinstance(
+                        prediction_metric,
+                        mm_schemas.ModelEndpointMonitoringMetricNoData,
+                    ):
```
Comment (jond01) [inline mlrun/model_monitoring/controller.py]: I suggest this more direct check:
```suggestion
                    if not prediction_metric.data:
```

File: mlrun/model_monitoring/controller.py [hunk 6]
```diff
@@ -443,154 +418,71 @@ def model_endpoint_process(
                 )
 
                 for start_infer_time, end_infer_time in batch_window.get_intervals():
-                    # start - TODO : delete in 1.9.0 (V1 app deprecation)
-                    try:
-                        # Get application sample data
-                        offline_response = cls._get_sample_df(
-                            feature_set=m_fs,
+                    prediction_metric = tsdb_connector.read_predictions(
```
Comment (jond01) [inline mlrun/model_monitoring/controller.py:418]: I'm somewhat worried about this frequent check, especially with TDEngine - see [ML-7806](https://iguazio.atlassian.net/browse/ML-7806).
Is there a way to remove/reduce this read operation?

File: mlrun/model_monitoring/controller.py [hunk 7]
```diff
@@ -443,154 +418,71 @@ def model_endpoint_process(
...
-                    feature_stats = json.loads(
-                        endpoint[mm_constants.EventFieldType.FEATURE_STATS]
-                    )
-
-                    # Pad the original feature stats to accommodate current
-                    # data out of the original range (unless already padded)
-                    pad_features_hist(FeatureStats(feature_stats))
-
-                    # Get the current stats:
-                    current_stats = calculate_inputs_statistics(
```
Comment (jond01) [inline mlrun/model_monitoring/controller.py:491]: Following these changes, is the controller still CPU-bound?
It seems to do only IO. We should remove the multi-process usage or replace it with async/multi-threading in this case.

**Review by jond01 — COMMENTED** [review]
Great removal. We should discuss the controller's design

File: mlrun/db/httpdb.py (hunk 4 above)
Comment (davesh0812) [reply mlrun/db/httpdb.py]: already in dev #6341

File: mlrun/model_monitoring/controller.py (hunk 6 above)
Comment (davesh0812) [reply mlrun/model_monitoring/controller.py:418]: Do you have different suggestion?

File: mlrun/model_monitoring/controller.py (hunk 7 above)
Comment (davesh0812) [reply mlrun/model_monitoring/controller.py:491]: Lets discuss offline 👍

File: mlrun/model_monitoring/controller.py (hunk 6 above)
Comment (jond01) [reply mlrun/model_monitoring/controller.py:418]: Let's keep it for now

File: mlrun/model_monitoring/controller.py (hunk 7 above)
Comment (jond01) [reply mlrun/model_monitoring/controller.py:491]: So just need to change `concurrent.futures.ProcessPoolExecutor` -> `concurrent.futures.ThreadPoolExecutor`

File: tests/system/model_monitoring/test_app.py [hunk 8]
```diff
@@ -847,7 +804,7 @@ def custom_setup_class(cls) -> None:
 
         # model monitoring app
         cls.app_data = _AppData(
-            class_=NoCheckDemoMonitoringApp, rel_path="assets/application.py"
+            class_=DemoMonitoringApp, rel_path="assets/application.py"
```
Comment (jond01) [inline tests/system/model_monitoring/test_app.py]: As we discussed
```suggestion
            class_=NoCheckDemoMonitoringApp, rel_path="assets/application.py", results={"data_drift_test", "model_perf"}
```

File: docs/tutorials/05-model-monitoring.ipynb [hunk 9]
```diff
@@ -552,7 +552,7 @@
    "id": "c3ce647b",
    "metadata": {},
    "source": [
-   "The **Model Monitoring Performance** dashboard displaysdrift and operational metrics over time."
+    "The **Model Monitoring Performance** dashboard displaysdrift and operational metrics over time."
```
Comment (jond01) [inline docs/tutorials/05-model-monitoring.ipynb]:
```suggestion
    "The **Model Monitoring Performance** dashboard displays drift and operational metrics over time."
```

**Review by jond01 — APPROVED** [review]

**Review by assaf758 — APPROVED** [review]

**Pull request merged**